Assets/Scripts/World/Units/Enemy/Main/EnemyController.cs

```diff
@@ -108,7 +111,28 @@ public class EnemyController : UnitController<IEnemy, Enemy, EnemyList>, IEnemyC
 		Instance = null;
 	}
 
-	public void Create(Enemy unit) {
+    public void AddActiveEnemy(EnemyBehaviour enemy) {
+        if (!activeEnemies.Contains(enemy)) {
+            activeEnemies.Add(enemy);
+        }
+    }
+
+    public void RemoveActiveEnemy(EnemyBehaviour enemy) {
+        if (activeEnemies.Contains(enemy)) {
+            activeEnemies.Remove(enemy);
+        }
+    }
+
+    public void KillAllEnemies() {
+        for (int i = 0; i < activeEnemies.Count; i++) {
+            int EnemyMaxHealth = activeEnemies.ElementAt(i).IMaxHealth;
+
+            activeEnemies.ElementAt(i).Heal(-EnemyMaxHealth);
```

Comment:
Sorry I missed this on my initial review, this is a very creative way of implementing. However, it may be more straightforward to simply call activeEnemies.ElementAt(i).Destroy()